Shopify/cli
# Deploy functions with unified app deployments when beta is enabled

### WHY are these changes introduced?
Functions are being updated to be deployed through unified app deployments. This PR makes that possible as long as the beta flag is set.

### WHAT is this pull request doing?
When the beta is enabled, we deploy functions through the `CreateDeployment` mutation. Otherwise, deploy functions in the old way.

### How to test your changes?

You'll need my `js.deploy-functions-on-ef` Partners branch and `js.add-module-id-to-upload-details` Script Service branch.

Two notes:
1. There is currently no way to migrate a function outside of unified app deployments to a function on unified app deployments, but that change will come later. This means if you want to switch between the two, you'll have to remove your `.env` file.
2. Functions deployed through unified app deployments wont actually run on the server yet. We still have to hook that up, but that will not change this work. 

Testing:
1. enable the unified app deployments beta on your Partners app
2. create a function (if you haven't already)
3. deploy, make sure it succeeds

### Checklist

- [X] I've considered possible cross-platform impacts (Mac, Linux, Windows) - No impact.
- [X] I've considered possible [documentation](https://shopify.dev) changes - None needed yet.
- [ ] I've made sure that any changes to `dev` or `deploy` have been reflected in the [internal flowchart](https://www.figma.com/file/7vqUp50u6dm48Zfb4JRRn8/CLI3-Internals) - Not yet, but I can
